datafusion/functions/src/string/upper.rs

```diff
@@ -59,6 +59,49 @@ impl ScalarUDFImpl for UpperFunc {
     }
 
     fn invoke(&self, args: &[ColumnarValue]) -> Result<ColumnarValue> {
-        handle(args, |string| string.to_uppercase(), "upper")
+        case_conversion(args, |string| string.to_uppercase(), "upper")
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use arrow::array::{ArrayRef, StringArray};
+    use std::sync::Arc;
+
+    #[test]
+    fn upper() -> Result<()> {
+        let string_array = Arc::new(StringArray::from(vec![
+            Some("arrow"),
+            None,
+            Some("datafusion"),
```

Comment:
Can you please also add tests for non ascii characters whose upper/lowercase lengths are different?

It seems like this is possible
https://stackoverflow.com/questions/48351343/can-lowercasing-a-utf-8-string-cause-it-to-grow